File: app/src/main/java/de/baumann/browser/Browser/BrowserContainer.java
package de.baumann.browser.Browser;

import de.baumann.browser.View.NinjaWebView;

import java.util.LinkedList;
import java.util.List;

public class BrowserContainer {
    private static final List<AlbumController> list = new LinkedList<>();

    public static AlbumController get(int index) {
        return list.get(index);
    }

    public synchronized static void set(AlbumController controller, int index) {
        if (list.get(index) instanceof NinjaWebView) {
            ((NinjaWebView) list.get(index)).destroy();
        }

        list.set(index, controller);
    }

    public synchronized static void add(AlbumController controller) {
        list.add(controller);
    }

    public synchronized static void add(AlbumController controller, int index) {
        list.add(index, controller);
    }

    public synchronized static void remove(AlbumController controller) {
        if (controller instanceof NinjaWebView) {
            ((NinjaWebView) controller).destroy();
        }

        list.remove(controller);
    }

    public static int indexOf(AlbumController controller) {
        return list.indexOf(controller);
    }

    public static List<AlbumController> list() {
        return list;
    }

    public static int size() {
        return list.size();
    }

    public synchronized static void clear() {
        for (AlbumController albumController : list) {
            if (albumController instanceof NinjaWebView) {
                ((NinjaWebView) albumController).destroy();
            }
        }

        list.clear();
    }
}

File: app/src/main/java/de/baumann/browser/browser/BrowserContainer.kt
package de.baumann.browser.browser

import kotlin.jvm.Synchronized
import de.baumann.browser.view.NinjaWebView
import java.util.*

class BrowserContainer {
    private val list: MutableList<AlbumController> = LinkedList()
    operator fun get(index: Int): AlbumController {
        return list[index]
    }

    @Synchronized
    fun add(controller: AlbumController) = list.add(controller)

    @Synchronized
    fun add(controller: AlbumController, index: Int) = list.add(index, controller)

    @Synchronized
    fun remove(controller: AlbumController) {
        (controller as NinjaWebView).destroy()
        list.remove(controller)
    }

    fun indexOf(controller: AlbumController?): Int = list.indexOf(controller)

    fun list(): List<AlbumController> = list

    fun size(): Int = list.size

    @Synchronized
    fun clear() {
        for (albumController in list) {
            (albumController as NinjaWebView).destroy()
        }
        list.clear()
    }
}